dotnet/src/Microsoft.Agents.AI.Hosting/HostedAgentBuilderExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Agents.AI.Hosting;
+
+/// <summary>
+/// Provides extension methods for configuring <see cref="AIAgent"/>.
+/// </summary>
+public static class HostedAgentBuilderExtensions
+{
+    /// <summary>
+    /// Configures the host agent builder to use an in-memory thread store for agent thread management.
+    /// </summary>
+    /// <param name="builder">The host agent builder to configure with the in-memory thread store.</param>
+    /// <returns>The same <paramref name="builder"/> instance, configured to use an in-memory thread store.</returns>
+    public static IHostedAgentBuilder WithInMemoryThreadStore(this IHostedAgentBuilder builder)
```

Comment:
This should be `AddInMemoryThreadStore`. Make sure to use existing naming conventions when creating new APIs. As far as I'm aware, there are no IHostedAgentBuilder.With* APIs today.

Reply:
renamed. I was trying to follow what Aspire has (`WithUrls`, `WithEndpoint`), but I dont have a strong opinion.